tests/MessagePack.Tests/MessagePackReaderTests.ReadString.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Buffers;
+using Xunit;
+
+namespace MessagePack.Tests
+{
+    partial class MessagePackReaderTests
+    {
+        [Fact]
+        public void ReadString_HandlesSingleSegment()
+        {
+            var seq = BuildSequence(new[] {
+                (byte)(MessagePackCode.MinFixStr + 2),
+                (byte)'A', (byte)'B' });
+
+            var reader = new MessagePackReader(seq);
+            var result = reader.ReadString();
+            Assert.Equal("AB", result);
+        }
+
+        [Fact]
+        public void ReadString_HandlesMultipleSegments()
+        {
+            var seq = BuildSequence(
+                new[] { (byte)(MessagePackCode.MinFixStr + 2), (byte)'A' },
+                new[] { (byte)'B' });
+
+            var reader = new MessagePackReader(seq);
+            var result = reader.ReadString();
+            Assert.Equal("AB", result);
```

Comment:
Without the fix, this assertion fails, because `result` equals `"AA"` instead of `"AB"`.